docs/en/engines/table-engines/mergetree-family/annindexes.md

```diff
@@ -0,0 +1,107 @@
+# Approximate Nearest Neighbor Search Indexes [experimental] {#table_engines-ANNIndex}
+
+The main task that indexes help to solve is to find the nearest neighbors for multidimensional data. An example of such a problem could be similar pictures or texts, for which the problem is reduced to finding the nearest [embeddings](https://cloud.google.com/architecture/overview-extracting-and-serving-feature-embeddings-for-machine-learning). They can be created from data using [UDF](../../../sql-reference/functions/index.md#executable-user-defined-functions).
+
+Next query can find closest neighbor in L2 space:
+``` sql 
+SELECT * 
+FROM table_name 
+WHERE L2Distance(Column, TargetEmbedding) < Value 
+LIMIT N
+```
+But it will take some time for execution because of the long calculation of the distance between `TargetEmbedding` and all other vectors. This is where indexes can help. As they  store the overall data structure using some methods (clustering, building search trees, etc.), they can give only approximate results for finding the nearest neighbors.
+
+## Indexes Structure
+
+Approximate Nearest Neighbor Search Indexes (`ANNIndexes`) are simmilar to skip indexes. They are constructed by some granules and determine which of them should be skipped. Compared to skip indices, ANN indices use their results not only to skip some group of granules, but also to select particular granules from a set of granules.
+
+`ANNIndexes` are designed to speed up two types of queries:
+
+- ######  Type 1: Where 
+   ``` sql 
+   SELECT * 
+   FROM table_name 
+   WHERE DistanceFunction(Column, TargetEmbedding) < Value 
+   LIMIT N
+   ```
+- ###### Type 2: Order by
+  ``` sql
+  SELECT * 
+  FROM table_name [WHERE ...] 
+  ORDER BY DistanceFunction(Column, TargetEmbedding) 
+  LIMIT N
+  ```
+
+In these queries, `DistanceFunction` is selected from [tuples of distance functions](../../../sql-reference/functions/tuple-functions/#l1norm). `TargetEmbedding` is a known embedding (something like `(0.1, 0.1, ... )`). `Value` - a float value that will bound the neighbourhood.
+
+!!! note "Note"
+    ANNIndex can't speed up query that satisfies both types(`where + order by`, only one of them). All queries must have the limit, as algorithms are used to find nearest neighbors and need a specific number of them.
```

Comment:
Yes. We have no logic for these queries because the indexes are trying to find kNN problems. Order by with k limit = find k neighbors. Where distance < value limit k will take k neighbors from the index and check their distances to the object to select granules.  It might be worth describing the limit value for where query more precisely in the documentation.
We could add support for this type, but the problem is that indexes would then work for where, but the order by part is more natural for indexes (because it's a nearest neighbor task)